### PR TITLE
Update webapp treasury viewer to show Base treasury and multi-sig balances

### PR DESCRIPTION
**Pull request by r4topunk**
I have updated it in 2 parts, the first one using the useBalance hooks, e the second one creating a new functions to get a list of token balances from a wallet.

I don't know if I liked the code I wrote, but I'm available to update the code with some tips.

I have added Base to Wagmi config, but maintain Mainnet while we are in the migration.

And another thing I think I'm not right, is the update in the sdk file, I just updated the file directly, but I think this file is generated and this is not the right approach.

Anyway, like I said, I'm willing to change the code and try again.

This code resolves the #21 issue

Captura de Tela 2024-04-13 às 19 47 07

### CODE REVIEW

File: packages/webapp/constants/gnarsDao.ts [hunk 1]
```diff
@@ -6,3 +6,10 @@ export const TREASURY_ADDRESS = "0xa1b74d2280966a89ac7e0f3a8bc5f0867c776d98"
 export const MULTISIG_ADDRESS = "0x0658f4eD17289144717713ADfFC2539eF7c2EF8e"
 export const USDC_TOKEN_ADDRESS = "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"
 export const V2_START_ID = 627
+
+export const BASE_TREASURY_ADDRESS = "0x72ad986ebac0246d2b3c565ab2a1ce3a14ce6f88"
```
Comment (bagelface) [inline packages/webapp/constants/gnarsDao.ts:10]: would recommend using the checksum version of all of these addresses

File: packages/webapp/utils/sdk.ts [hunk 2]
```diff
@@ -2138,10 +2138,10 @@ export const gnarsV2TokenABI = [
 ] as const
 
 /**
- * [__View Contract on Ethereum Etherscan__](https://etherscan.io/address/0x558BFFF0D583416f7C4e380625c7865821b8E95C)
+ * [__View Contract on Ethereum BaseScan__](https://basescan.org/address/0x880Fb3Cf5c6Cc2d7DFC13a993E839a9411200C17)
```
Comment (bagelface) [inline packages/webapp/utils/sdk.ts:2141]: Was all this manually updated or was it generated using `yarn wagmi generate`?

File: packages/webapp/utils/web3.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ * Fetch tokens for a wallet
+ * @param holder token holder address
+ * @param tokens list of tokens to find
+ * @returns the data of the tokens
+ */
+export function getTokensValues(holder: `0x${string}`, tokens: `0x${string}`[]) {
+  const ethBalancePromise = fetchBalance({ address: holder })
+  const balancePromises = tokens.map((token) => fetchBalance({ address: holder, token }))
+  return Promise.all([ethBalancePromise, ...balancePromises])
+}
```
Comment (bagelface) [inline packages/webapp/utils/web3.ts:13]: line break at the end of this file

File: packages/webapp/utils/web3.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { fetchBalance } from "wagmi/actions"
+
+/**
+ * Fetch tokens for a wallet
+ * @param holder token holder address
+ * @param tokens list of tokens to find
+ * @returns the data of the tokens
+ */
+export function getTokensValues(holder: `0x${string}`, tokens: `0x${string}`[]) {
```
Comment (bagelface) [inline packages/webapp/utils/web3.ts:9]: not sure what the convention is throughout this project, but always nice to include the return type

**Review by bagelface — APPROVED** [review]
Everything here looks solid to me. Code is nice and clean and everything works as expected. Dropped a few small notes but, none of them would block this from being merge imo.

**Comment by r4topunk** [comment]
@bagelface I will update the code with your considerations, then I will ask for another review then I will ask for merge

**Review by mulfdev — CHANGES_REQUESTED** [review]
Per out tech pod meeting today, lets integrate the token formatting and wallet/chain agnostic data fetching. great work!

**Comment by sktbrd** [comment]
Looks good to me. @bagelface @mulfdev what you think ?

**Pull request merged**